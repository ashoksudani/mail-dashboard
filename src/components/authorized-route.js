import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Route, Redirect } from 'react-router-dom';

import * as selectors from 'selectors';

export class AuthorizedRoute extends Component {
  render() {
    const { component: Component, profile, ...rest } = this.props;
    return (
      <Route
        {...rest}
        render={props => {
          if (profile.isAuthenticated) {
            return <Component {...props} />;
          } else {
            return <Redirect to="/login" />;
          }
        }}
      />
    );
  }
}

const mapStateToProps = (state, props) => {
  return {
    profile: selectors.selectProfile(state)
  };
};

export default connect(mapStateToProps)(AuthorizedRoute);
