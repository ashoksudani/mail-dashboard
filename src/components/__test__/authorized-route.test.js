import React from 'react';
import { AuthorizedRoute } from '../authorized-route';
import { Redirect, MemoryRouter } from 'react-router-dom';

test('renders passed component of route if profile is authenticated', () => {
  const SampleComp = () => <div id="sample-div">Sample Div</div>;
  const props = {
    component: SampleComp,
    path: '/login',
    profile: { isAuthenticated: true }
  };

  const rootWrapper = mount(
    <MemoryRouter initialEntries={['/login']} initialIndex={0}>
      <AuthorizedRoute {...props} />
    </MemoryRouter>
  );

  expect(rootWrapper).toMatchSnapshot();

  expect(rootWrapper.find('div[id="sample-div"]').length).toEqual(1);
});

test('renders passed component of route if profile is not authenticated', () => {
  const SampleComp = () => <div id="sample-div">Sample Div</div>;
  const props = {
    component: SampleComp,
    path: '/mail',
    profile: { isAuthenticated: false }
  };

  const rootWrapper = shallow(
    <MemoryRouter initialEntries={['/mail']} initialIndex={0}>
      <AuthorizedRoute {...props} />
    </MemoryRouter>
  );
  expect(rootWrapper.html()).toEqual('');
});
