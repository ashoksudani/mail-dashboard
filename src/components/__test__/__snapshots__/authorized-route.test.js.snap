// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders passed component of route if profile is authenticated 1`] = `
<div
  id="sample-div"
>
  Sample Div
</div>
`;
